sdl_android/src/main/java/com/smartdevicelink/api/screen/TextAndGraphicManager.java

```diff
@@ -0,0 +1,893 @@
+				}
+			} else {
+				// If text 3 does not exist, put it in slot 3 formatted
+				tempString.append(textField4);
+				if (textField4Type != null){
+					tags.setMainField3(textField4Type);
+				}
+			}
+		}
+
+		show.setMainField3(tempString.toString());
+		show.setMetadataTags(tags);
+		return show;
+	}
+
+	private Show assembleFourLineShowText(Show show){
+
+		MetadataTags tags = new MetadataTags();
+
+		if (textField1 != null && textField1.length() > 0) {
+			show.setMainField1(textField1);
+			if (textField1Type != null){
+				tags.setMainField1(textField1Type);
+			}
+		}
+
+		if (textField2 != null && textField2.length() > 0) {
+			show.setMainField2(textField2);
+			if (textField2Type != null){
+				tags.setMainField2(textField2Type);
+			}
+		}
+
+		if (textField3 != null && textField3.length() > 0) {
+			show.setMainField3(textField3);
+			if (textField3Type != null){
+				tags.setMainField3(textField3Type);
+			}
+		}
+
+		if (textField4 != null && textField4.length() > 0) {
+			show.setMainField4(textField4);
+			if (textField4Type != null){
+				tags.setMainField4(textField4Type);
+			}
+		}
+
+		show.setMetadataTags(tags);
+		return show;
+	}
+
+	// Extraction
+
+	protected Show extractTextFromShow(Show show){
+
+		Show newShow = new Show();
+		newShow.setMainField1(show.getMainField1());
+		newShow.setMainField2(show.getMainField2());
+		newShow.setMainField3(show.getMainField3());
+		newShow.setMainField4(show.getMainField4());
+
+		return newShow;
+	}
+
+	private Show setBlankTextFields(Show newShow){
+
+		newShow.setMainField1("");
+		newShow.setMainField2("");
+		newShow.setMainField3("");
+		newShow.setMainField4("");
+		newShow.setMediaTrack("");
+
+		return newShow;
+	}
+
+	private void updateCurrentScreenDataFromShow(Show show){
+
+		if (show == null){
+			Log.e(TAG, "can not updateCurrentScreenDataFromShow from null show");
+			return;
+		}
+
+		// If the items are null, they were not updated, so we can't just set it directly
+		if (show.getMainField1() != null){
+			currentScreenData.setMainField1(show.getMainField1());
+		}
+		if (show.getMainField2() != null){
+			currentScreenData.setMainField2(show.getMainField2());
+		}
+		if (show.getMainField3() != null){
+			currentScreenData.setMainField3(show.getMainField3());
+		}
+		if (show.getMainField4() != null){
+			currentScreenData.setMainField4(show.getMainField4());
+		}
+		if (show.getMediaTrack() != null){
+			currentScreenData.setMediaTrack(show.getMediaTrack());
+		}
+		if (show.getMetadataTags() != null){
+			currentScreenData.setMetadataTags(show.getMetadataTags());
+		}
+		if (show.getAlignment() != null){
+			currentScreenData.setAlignment(show.getAlignment());
+		}
+		if (show.getGraphic() != null){
+			currentScreenData.setGraphic(show.getGraphic());
+		}
+		if (show.getSecondaryGraphic() != null){
+			currentScreenData.setSecondaryGraphic(show.getSecondaryGraphic());
+		}
+	}
+
+	// Helpers
+
+	private List<String> findNonNullMainTextFields(){
```

Comment:
Since this actually returns if the values are not null along with if they are beyond empty set I would recommend changing the name to `findValidMainTextFields`